src/util/queryParams.js

```diff
@@ -0,0 +1,30 @@
+import qs from 'qs';
+
+export function getQueryParameters(queryString) {
+  let gistId = null;
+  let snapshotKey = null;
+  let isExperimental = false;
+  if (queryString) {
+    const query = qs.parse(queryString.slice(1));
+    if (query.gist) {
+      gistId = query.gist;
+    }
+    if (query.snapshot) {
+      snapshotKey = query.snapshot;
+    }
+    isExperimental = Object.keys(query).includes('experimental');
+  }
+  return {
+    gistId,
+    snapshotKey,
+    isExperimental,
+  };
+}
+
+export function setQueryParameters(isExperimental) {
```

Comment:
Even though `isExperimental` is the only possible parameter, I think it would feel better for this to take a params object, since `setQueryParameters(true)` would be a pretty inscrutable function call.